trx-request/src/lib.rs

```diff
@@ -68,13 +72,28 @@ fn parse_max_amount<'a>(t: &Token) -> Result<MaxAmount, ParseError<'a>> {
 
 fn parse_chain<'a>(chain: &'a str) -> Result<Chain, ParseError<'a>> {
     match chain {
+        "Gate" => Ok(Chain::Gate),
         "Eth" => Ok(Chain::Eth),
         _ => Err(ParseError::InvalidChain(chain)),
     }
 }
 
+fn parse_gate_address<'a>(account: &'a str) -> Result<[u8; 32], ParseError<'a>> {
+    // TODO: handle SS58 encoded addresses as well
+    if let Some(stripped) = account.strip_prefix("0x") {
+        let decoded: Vec<u8> =
+            hex::decode(stripped).map_err(|_| ParseError::InvalidChainAccount(Chain::Gate))?;
+        let address: [u8; 32] = decoded
+            .try_into()
+            .map_err(|_| ParseError::InvalidChainAccount(Chain::Gate))?;
+        Ok(address)
+    } else {
+        Err(ParseError::InvalidChainAccount(Chain::Gate))?
+    }
+}
+
 fn parse_eth_address<'a>(account: &'a str) -> Result<[u8; 20], ParseError<'a>> {
-    if &account[0..2] != "0x" {
+    if account.len() < 2 || &account[0..2] != "0x" {
```

Comment:
This should use starts_with, right?

Reply:
This one is a little different bc of the clauses below, we should maybe do a larger refactor but didn't want to do that yet